prog/vnet/load_balancer_health_probes.rb

```diff
@@ -8,18 +8,24 @@ def vm
   end
 
   label def health_probe
-    response_code = begin
+    address_ipv4 = load_balancer.ipv4_enabled? ? vm.nics.first.private_ipv4.network : nil
+    address_ipv6 = load_balancer.ipv6_enabled? ? vm.ephemeral_net6.nth(2) : nil
+
+    response_codes = [address_ipv4, address_ipv6].compact.map do |address|
       cmd = if load_balancer.health_check_protocol == "tcp"
-        "sudo ip netns exec #{vm.inhost_name} nc -z -w #{load_balancer.health_check_timeout} #{vm.nics.first.private_ipv4.network} #{load_balancer.dst_port} && echo 200 || echo 400"
+        "sudo ip netns exec #{vm.inhost_name} nc -z -w #{load_balancer.health_check_timeout} #{address} #{load_balancer.dst_port} && echo 200 || echo 400"
       else
-        "sudo ip netns exec #{vm.inhost_name} curl --insecure --resolve #{load_balancer.hostname}:#{load_balancer.dst_port}:#{vm.nics.first.private_ipv4.network} --max-time #{load_balancer.health_check_timeout} --silent --output /dev/null --write-out '%{http_code}' #{load_balancer.health_check_protocol}://#{load_balancer.hostname}:#{load_balancer.dst_port}#{load_balancer.health_check_endpoint}"
+        "sudo ip netns exec #{vm.inhost_name} curl --insecure --resolve #{load_balancer.hostname}:#{load_balancer.dst_port}:#{(address.version == 6) ? "[#{address}]" : address} --max-time #{load_balancer.health_check_timeout} --silent --output /dev/null --write-out '%{http_code}' #{load_balancer.health_check_protocol}://#{load_balancer.hostname}:#{load_balancer.dst_port}#{load_balancer.health_check_endpoint}"
+      end
+      begin
+        vm.vm_host.sshable.cmd(cmd).strip
+      rescue
+        "500"
       end
-
-      vm.vm_host.sshable.cmd(cmd)
-    rescue
-      "500"
     end
 
+    response_code = response_codes.include?("200") ? 200 : 400
```

Comment:
nit: maybe remove this line and below replace `response_code.to_i == 200 with` `response_codes.include?("200")`.